lib/nncp-bm/ocp_node_template.yaml

```diff
@@ -0,0 +1,88 @@
+---
+apiVersion: nmstate.io/v1
+kind: NodeNetworkConfigurationPolicy
+metadata:
+  name: _ignored_
+spec:
+  desiredState:
+    interfaces:
+      - description: internalapi vlan interface
+        ipv4:
+          address:
+            - ip: _replaced_
+              prefix-length: _replaced_
+          enabled: true
+          dhcp: false
+        ipv6:
+          enabled: false
+        name: internalapi
+        state: up
+        type: vlan
+        vlan:
+          base-iface: _replaced_
+          id: _replaced_
+        mtu: 1500
+      - description: storage vlan interface
+        ipv4:
+          address:
+            - ip: _replaced_
+              prefix-length: _replaced_
+          enabled: true
+          dhcp: false
+        ipv6:
+          enabled: false
+        name: storage
+        state: up
+        type: vlan
+        vlan:
+          base-iface: _replaced_
+          id: _replaced_
+        mtu: 1500
+      - description: tenant vlan interface
+        ipv4:
+          address:
+            - ip: _replaced_
+              prefix-length: _replaced_
+          enabled: true
+          dhcp: false
+        ipv6:
+          enabled: false
+        name: tenant
+        state: up
+        type: vlan
+        vlan:
+          base-iface: _replaced_
+          id: _replaced_
+        mtu: 1500
+      - description: ctlplane interface
+        name: _replaced_
+        state: up
+        type: ethernet
+        mtu: 1500
+        ipv6:
+          enabled: false
+        ipv4:
+          address:
+            - ip: _replaced_
+              prefix-length: _replaced_
+          enabled: true
+          dhcp: false
+      - description: external vlan interface
+        ipv4:
+          address:
+            - ip: _replaced_
+              prefix-length: _replaced_
+          enabled: true
+          dhcp: false
+        ipv6:
+          enabled: false
+        name: external
+        state: up
+        type: vlan
+        vlan:
+          base-iface: _replaced_
+          id: _replaced_
+        mtu: 1500
```

Comment:
Are you creating this new `lib/nncp-bm` content because you need to add this `external` interface?  If so, I think you might be able to do it with `patches` and reuse the `lib/nncp` content.

Reply:
See this for an example of what I'm describing: https://github.com/openstack-k8s-operators/architecture/blob/7d67cefda341105d64ed19a2cad05d749a493d52/dt/uni07eta/nncp/kustomization.yaml#L22-L36

Reply:
In our case, the name of the master node is variable based on the allocation we are getting in scale lab. if so how to add the external network on each machine's nncp. do you have any pointer for that. thanks!

Reply:
I see what you're saying.  Since [1] becomes the name of the first node in the `values.yaml`, then the patch needs to target that same name.  And since you don't know that name, you can't hardcode it here.  I'll have to think if there's a way to do this dynamically.

[1] https://github.com/openstack-k8s-operators/architecture/blob/a596f2cff1adf47008566a3760ddcdbafda7c8ef/lib/nncp/ocp_nodes_nncp.yaml#L5

Reply:
Ah, wait.  I have an idea.  Since you're adding this same hardcoded template to all `NNCP`s...

```
      - description: external vlan interface
        ipv4:
          address:
            - ip: _replaced_
              prefix-length: _replaced_
          enabled: true
          dhcp: false
        ipv6:
          enabled: false
        name: external
        state: up
        type: vlan
        vlan:
          base-iface: _replaced_
          id: _replaced_
        mtu: 1500
```

...you can just change the patch `target` and remove the `name`:

```

 patches: 
   - target: 
       kind: NodeNetworkConfigurationPolicy 
     patch: |-
     ...
```

Or maybe have to use `name: .*`.  One or the other should work though, I think.